pkg/pillar/cmd/zedmanager/zedmanager.go

```diff
@@ -467,8 +505,26 @@ func handleCreate(ctxArg interface{}, key string,
 			// FIXME - We really need a top level flag to tell the app is
 			//  a container. Deriving it from Storage seems hacky.
 			status.IsContainer = true
+			sha := utils.MaybeNameHasSha(sc.Name)
+			if sha != "" {
+				ss.ImageSha256 = sha
+				addAppAndImageHash(ctx, config.UUIDandVersion.UUID,
+					ss.ImageID, ss.ImageSha256)
+			} else {
+				log.Infof("Setting waitForResolve to true before publishing resolve config")
+				waitForResolve = true
+				status.RefCount++
+				publishAppInstanceStatus(ctx, &status)
+				resolveConfig := types.ResolveConfig{
+					DatastoreID: sc.DatastoreID,
+					Name:        sc.Name,
+					AllowNonFreePort: types.AllowNonFreePort(*ctx.globalConfig,
+						types.AppImgObj),
+					Counter: config.PurgeCmd.Counter,
+				}
+				publishResolveConfig(ctx, &resolveConfig)
+			}
 		}
-		// XXX before latching we should ResolveConfig to get the sha
 		maybeLatchImageSha(ctx, config, ss)
```

Comment:
I think this call should be deleted - no point in doing it until it has been resolved and I think you have a latch call when you handle the ResolverStatus

Reply:
Yes, I am latching it while handling resolve status. I removed it from here.